src/components/Input/Input.ts

```diff
@@ -0,0 +1,64 @@
+import Block from '../../modules/Block';
+import {InputProps} from './types';
+import {template} from './template';
+import {TValidateResult} from '../../modules/Validate/types';
+import {InputName} from '../../shared/const';
+import validator from '../../modules/Validate';
+import {TRenderElement} from '../../modules/Block/types';
+
+class Input extends Block<InputProps> {
+  constructor(props:InputProps) {
+    super( {
+      isValid: true,
+      inputValue: '',
+      events: {
+        focusin: ()=>this.onFocus(),
+        focusout: (e:FocusEvent)=>this.onBlur(e),
+      },
+      ...props,
+    });
+  }
+  onValidate(name:InputName, value:FormDataEntryValue | null, compareValue?:string):TValidateResult | void {
+    if (typeof value !== 'string') return;
+    return validator[name](value, compareValue);
+  }
+
+  onUpdate(name:InputName, value: FormDataEntryValue | null, compareValue?:string):void {
```

Comment:
Я бы порекомендовал делать отступы у типов
```suggestion
  onUpdate(name: InputName, value: FormDataEntryValue | null, compareValue?: string) : void {
```